docs/sections/user_guide/rocoto_yaml.rst

```diff
@@ -0,0 +1,606 @@
+
+
+Workflow Section
+~~~~~~~~~~~~~~~~
+Starting at the top level of the YAML config for Rocoto, there are several required sections. See the example and explanation below:
+
+.. code::
+
+  workflow:
+    attrs:
+      realtime: false
+      scheduler: slurm
+    cycledef:
+      - attrs:
+          activation_offset: -06:00
+          group: howdy
+        spec: 202209290000 202209300000 06:00:00
+    entities:
+      ACCOUNT: myaccount
+      FOO: test.log
+    log: /some/path/to/&FOO;
+    tasks:
+      ...
+
+YAML Entries
+............
+
+``attrs``: This section allows users to define any number of the available attributes available to the workflow tag in Rocoto's native XML language. Set them with a key/value that matches the exact syntax needed by Rocoto. These attributes work to fill in the ``<workflow>`` tag. For example,
+
+.. code:: XML
+
+  <workflow realtime="false" scheduler="slurm">
+  ...
+  </workflow>
+
+``cycledef``: This section is comprised of a list of grouped cycle definitions for a workflow. Any number of entries is supported. Similar to ``attrs`` for the ``workflow`` level, this entry, too has an ``attrs`` tag that follows the exact requirements of those in the Rocoto XML language. The ``spec`` entry is required and supports either the "start, stop, step" syntax, or the "crontab-like" method supported by Rocoto. The example above translates to a single ``<cycledef>`` tag:
+
+.. code:: XML
+
+  <cycledef group="howdy">202209290000 202209300000 06:00:00</cycledef>
+
+``entities``: This section is a set of key/value pairs where the keys are the named entities (or XML variables) and the values track to the values of those XML entities. Each entry shows up as an entry like ``<!ENTITY key "value">``. The example above would yield:
+
+.. code:: XML
+
+  <?xml version='1.0' encoding='utf-8'?>
+  <!DOCTYPE workflow [
+    <!ENTITY ACCOUNT "myaccount">
+    <!ENTITY FOO "test.log">
+  ]>
+
+``log``: This is a path-like string that defines where to put the Rocoto logs. It corresponds to the ``<log>`` tag. For example:
+
+.. code:: XML
+
+  <log>/some/path/to/&FOO;</log>
+
+
+``tasks``: This section will be explained in detail in the next section.
+
+
+Using Cycle Strings
+...................
+
+The ``<cyclestr>`` tag in Rocoto transforms specific flags to represent components of the current cycle at run time. For example, an ISO date string like ``2023-01-01T12:00:00`` is represented as ``'@Y-@m-@dT@X'``. See the :rocoto:`Rocoto documentation<>` for full details. In the Rocoto YAML, the ``cyclestr:`` entry can be used anywhere that Rocoto will accept a ``<cyclestr>`` to achieve this result. The required structure of a ``cyclestr:`` entry is a ``value:``, like this:
```

Comment:
When editing these, it was obvious I started liking to add the visual cue that it was a YAML entry over time. I tried to standardize on having the colon. Let's hope I didn't miss any.

Reply:
I'm not sure if we have any discussion of specific YAML syntax on other pages yet, but it could be worth a quick scan for consistency. What do you think about adding this to the [guidelines page](https://unified-workflow.readthedocs.io/en/doc-update/sections/contributor_guide/documentation.html#documentation-guidelines)?